### DISCUSSION TITLE
How to add class to Figure element in Markdown

### DISCUSSION THREAD

**Discussion by ionm** [jgm/pandoc#9546] [Q&A] [answered]
Could you tell me how I can define the class of a Figure element in Markdown? I have noticed that `![Caption](image.jpg){.margin}` defines the class for `Image` and not for `Figure`.

I want to write a Lua filter to change the environment of certain images (selected by class) from `figure` to `marginfigure`.

**Comment by jgm** [comment, marked as answer]
Currently there's no way to do this.  You could write a Lua filter to move the class from the Image contained in a Figure to the Figure itself.

**Reply by jgm** [reply]
You might find it more convenient to wrap the figure in a fenced div with a class; then your filter could just look for these Divs and adjust the figures inside them.

**Comment by badlydrawnrob** [comment]
@jgm I've found myself in the position of needing this too! My particular needs are using it with flexbox. You'd either have to `#parent > figure` or as you say wrap with a `<div>` class. It's a bit of a shame this isn't possible.

**Reply by rnwst** [reply]
You might find the [`has`](https://developer.mozilla.org/en-US/docs/Web/CSS/:has) selector useful:
```css
figure:has(img.my-class) {
  display: flex;
}
```

**Reply by badlydrawnrob** [reply]
@rnwst A little ugly but it'll do 😉
